src/rasmus_fuel/realistic.py

```diff
@@ -98,6 +98,19 @@ def power_maintain_sog(
     speed_og = (u_ship_og ** 2 + v_ship_og ** 2) ** 0.5
     speed_tw = (u_ship_tw ** 2 + v_ship_tw ** 2) ** 0.5
     speed_rel_to_wind = ((u_ship_og - u_wind) ** 2 + (v_ship_og - v_wind) ** 2) ** 0.5
+    
+     # calc wind direction and speed
+    angle_wind = np.arctan2(v_wind, u_wind)
+    speed_wind = (u_wind ** 2 + v_wind ** 2) ** 0.5
+
+    # calc angle between wind and speed through water 
+    angle_ship_wind = np.arccos(
+        (u_wind * u_ship_tw  + v_wind * v_ship_tw) / speed_tw / wind_speed
+    )
+
+    # calc projection of ship velocity through water on wind components
+    u_speed_rel_to_wind = speed_tw * np.cos(angle_ship_wind) * np.cos(angle_wind)
+    v_speed_rel_to_wind = speed_tw * np.cos(angle_ship_wind) * np.sin(angle_wind)
```

Comment:
Not sure about this. Shouldn't this just be the eastward and northward components of inverse of the the apparent wind? So then, we'd have: 
```python
u_speed_rel_to_wind = u_ship_og - u_wind
v_speed_rel_to_wind = v_ship_og - v_wind
speed_rel_to_wind = (u_speed_rel_to_wind ** 2 + u_speed_rel_to_wind ** 2) ** 0.5
```